perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -212,6 +214,8 @@ public Group updateGroup(PerunSession sess, Group group) throws InternalErrorExc
 			try {
 				jdbc.update("update groups set name=?,modified_by=?, modified_by_uid=?, modified_at=" + Compatibility.getSysdate() + " where id=?", dbGroup.getName(),
 						sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getUserId(), dbGroup.getId());
+			} catch (DataIntegrityViolationException e) {
+				throw new GroupExistsException("The name must be unique and it's already occupied.", e);
```

Comment:
Maybe we should include used name in the exception text.

Reply:
I don't really think we need to know that. And it is exactly the same copy of the logic like in facilityManager.

Reply:
Do you insist? Of course I can add it, but I don't think it is important except the situation when database constraint works improperly.

Reply:
Not really, but since its in DAO layer, it might fail also some inner logic (move group?, group structure sync?). I'm not sure. Right now it won't be possible to know, which group caused the problem, unless it is logged anywhere else, since even passed SQL exception doesn't contain input data.

Reply:
Ok - added.